compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -332,14 +332,20 @@ impl<'tcx> Collector<'tcx> {
                 }
             }
 
-            if let Some((_, span)) = wasm_import_module {
+            if let Some((_, span)) = wasm_import {
```

Comment:
This can be moved into the `match` above now.

Reply:
The above `match` is in a loop, if we move this to it we need to emit the same diagnostic in several branches?

For code:
```rust
#[link(name = "foo", wasm_import_module = "bar")] 
``` 
and 
```rust
#[link(wasm_import_module = "bar", name = "foo")]
```

we both want to emit:
```console
`wasm_import_module` is incompatible with other arguments in `#[link]` attributes
```

Keep it out of the loop we may don't care the order?

Reply:
Ok, I see.